kubernetes/onpremise/kubeadm/scripts/install_kubernetes.sh

```diff
@@ -0,0 +1,177 @@
+  %{~ endif ~}
+  --node-name master
+
+# #Calico
+# sudo curl -s https://docs.projectcalico.org/manifests/calico.yaml > calico.yaml
+# sudo curl -s https://docs.tigera.io/calico/latest/manifests/calico.yaml > calico.yaml
+# sudo sed -i -e 's?# - name: CALICO_IPV4POOL_CIDR?- name: CALICO_IPV4POOL_CIDR?g' calico.yaml
+# sudo sed -i -e 's?#   value: "192.168.0.0/16"?  value: "192.168.0.0/16"?g' calico.yaml
+# sudo kubectl apply -f calico.yaml
+
+mkdir -p $HOME/.kube
+sudo cp /etc/kubernetes/admin.conf $HOME/.kube/config
+sudo chown $(id -u):$(id -g) $HOME/.kube/config
+
+# Prepare kubeconfig file for downloading
+#sudo chmod 660 /etc/kubernetes/admin.conf
+sudo cp /etc/kubernetes/admin.conf /tmp/admin.conf
+sudo chown ${user}:${user} /tmp/admin.conf
+chmod 660 /tmp/admin.conf
+kubectl --kubeconfig /tmp/admin.conf config set-cluster kubernetes --server "https://${master_public_ip}:6443"
+
+#------------------------------------------------------------------------------#
+# Master: Network Pluggin configuration
+#------------------------------------------------------------------------------#
+%{~ if cni_pluggin == "calico" ~}
+echo "----> config network : CNI pluggin will be Calico !"
+sudo curl -s  https://raw.githubusercontent.com/projectcalico/calico/v3.25.1/manifests/tigera-operator.yaml > tigera-operator.yaml
+sudo kubectl create -f tigera-operator.yaml
+sudo curl -s https://raw.githubusercontent.com/projectcalico/calico/v3.25.1/manifests/custom-resources.yaml > custom-resources.yaml
+sudo kubectl create -f custom-resources.yaml
+%{~ endif ~}
+
+%{~ if cni_pluggin == "flannel" ~}
+echo "----> config network : CNI pluggin will be Flannel !"
+kubectl apply -f https://github.com/flannel-io/flannel/releases/latest/download/kube-flannel.yml
+%{~ endif ~}
+
+#------------------------------------------------------------------------------#
+# Master: Install Load Balancer
+#------------------------------------------------------------------------------#
+%{~ if loadbalancer_plugin == "metalLB" ~}
+echo "Step 1 : preparation of metalLB !"
+# "metalLB config - Preparation step"
+kubectl get configmap kube-proxy -n kube-system -o yaml | \
+sed -e "s/strictARP: false/strictARP: true/" | \
+kubectl apply -f - -n kube-system
+# Step 2 : installation of metalLB :
+echo "Step 2 : installation of metalLB"
+kubectl apply -f https://raw.githubusercontent.com/metallb/metallb/v0.13.9/config/manifests/metallb-native.yaml
+# patch metal LB to be allowed to run on master node
+kubectl patch deployment controller -n metallb-system --type='json' -p='[{"op": "add", "path": "/spec/template/spec/tolerations", "value":[{"operator":"Exists","key":"node-role.kubernetes.io/control-plane","effect":"NoSchedule"}]}]'
+kubectl rollout restart deployment controller -n metallb-system
+
+#wait for metal lb pod to be ready
+while [[ $(kubectl get pods -n metallb-system -l app=metallb,component=controller -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}') != "True" ]];
```

Comment:
Please make it POSIX compliant:

```suggestion
while [ "$(kubectl get pods -n metallb-system -l app=metallb,component=controller -o 'jsonpath={..status.conditions[?(@.type=="Ready")].status}')" != "True" ];
```